### PR TITLE
apt-cacher-ng 0.9.1

### PR DESCRIPTION
**Pull request by yous**
Bump [apt-cacher-ng](https://www.unix-ag.uni-kl.de/~bloch/acng/) to 0.9.1.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?
- [ ] Have you built your formula locally prior to submission with `brew install <formula>`?
- [ ] Have you successfully ran `brew test <formula>` with your changes locally?

### CODE REVIEW

**Comment by DomT4** [comment]
Patch failed to apply. It may well have been merged upstream. There's a link in the formula to the upstream bug report so should be easy enough to check status there.

**Comment by yous** [comment]
I found the patch has been applied to the source of 0.9.1. Removed the patch.

**Comment by mikemcquaid** [comment]
Failing on 10.9: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/43111/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_apt_cacher_ng/

**Comment by yous** [comment]
Currently getting following error messages:

```
[ 30%] Building CXX object source/CMakeFiles/acngstuff.dir/aclogger.cc.o
cd /tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1/source && /usr/local/Library/ENV/4.3/clang++    -I/tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1 -I/tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1/include  -DNDEBUG    -D_FILE_OFFSET_BITS=64 -pthread -Wall -Wextra -Wno-unused-parameter -fvisibility-inlines-hidden -D_DARWIN_C_SOURCE  -std=gnu++11 -I/usr/local/Cellar/openssl/1.0.2g/include  -o CMakeFiles/acngstuff.dir/aclogger.cc.o -c /tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1/source/aclogger.cc
/tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1/source/meta.cc:712:22: error: no member named 'fmemopen' in the global namespace; did you mean 'freopen'?
   FILE* memStrm = ::fmemopen( (void*) pAscii, len, "r");
                   ~~^~~~~~~~
                     freopen
/usr/include/stdio.h:246:7: note: 'freopen' declared here
FILE    *freopen(const char * __restrict, const char * __restrict,
         ^
/tmp/apt-cacher-ng20160311-66404-18asf7l/apt-cacher-ng-0.9.1/source/meta.cc:712:32: error: cannot initialize a parameter of type 'const char *' with an rvalue of type 'void *'
   FILE* memStrm = ::fmemopen( (void*) pAscii, len, "r");
                               ^~~~~~~~~~~~~~
/usr/include/stdio.h:246:38: note: passing argument to parameter here
FILE    *freopen(const char * __restrict, const char * __restrict,
                                        ^
2 errors generated.
make[3]: *** [source/CMakeFiles/acngstuff.dir/meta.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [source/CMakeFiles/acngstuff.dir/all] Error 2
make[1]: *** [source/CMakeFiles/apt-cacher-ng.dir/rule] Error 2
make: *** [apt-cacher-ng] Error 2
```

But it seems that `fmemopen` is in glibc and isn't supported by OS X. Am I correct?

**Comment by DomT4** [comment]
As far as I'm aware that's correct. Various people have ported [fmemopen] (https://github.com/NimbusKit/MemoryMapping) or mimicked it using `funopen` from a cursory glance. I suspect upstream aren't aware they used non-portable code here, so it's worth reporting to them.

**Comment by yous** [comment]
I found a way to build this by disabling OpenSSL and using LibTomCrypt instead.

**Comment by yous** [comment]
Is this acceptable? Or should we stay on 0.8.8 to use OpenSSL?

File: Library/Formula/apt-cacher-ng.rb [hunk 1]
```diff
@@ -13,26 +13,14 @@ class AptCacherNg < Formula
...
-    # https://stackoverflow.com/questions/28955478/when-can-we-omit-the-return-type-in-a-c11-lambda
-    #
-    # Raised https://alioth.debian.org/tracker/index.php?func=detail&aid=315276&group_id=100566&atid=413109
-    # with upstream to address.
-    patch :DATA
-  end
-
   def install
     ENV.cxx11
+    ENV["TOMCRYPT_HOME"] = Formula["libtomcrypt"].prefix
```
Comment (DomT4) [inline Library/Formula/apt-cacher-ng.rb]: `opt_prefix`

File: Library/Formula/apt-cacher-ng.rb [hunk 2]
```diff
@@ -13,26 +13,14 @@ class AptCacherNg < Formula
   depends_on "cmake" => :build
   depends_on :osxfuse => :build
   depends_on "boost" => :build
-  depends_on "openssl"
+  depends_on "libtomcrypt" => :build
```
Comment (DomT4) [inline Library/Formula/apt-cacher-ng.rb]: Why only `:build`, out of interest?

File: Library/Formula/apt-cacher-ng.rb [hunk 3]
```diff
@@ -42,7 +30,9 @@ def install
       s.gsub!(/^LogDir: .*/, "LogDir: #{var}/log")
     end
 
-    system "cmake", ".", *std_cmake_args
+    args = std_cmake_args
+    args << "-DUSE_SSL=OFF"
```
Comment (DomT4) [inline Library/Formula/apt-cacher-ng.rb]: Maybe worth adding a notation that SSL support is supplied by `libtomcrypt` rather than binned entirely.

**Comment by yous** [comment]
Updated, sorry for delay!

**Comment by DomT4** [comment]
Oh, `libtomcrypt` was removed to the boneyard recently:

* https://github.com/Homebrew/legacy-homebrew/issues/43906.
* https://github.com/Homebrew/homebrew-core/pull/4.

So we'll need to use something else :cry:.

**Comment by yous** [comment]
Hmm... Actually I have no idea. We should use OpenSSL instead of LibTomCrypt again, then we have to go over the problem with `fmemopen`. But I'm not sure that it's a right approach porting `fmemopen` to make the package compatible with OS X.

**Comment by mikemcquaid** [comment]
Yeh, using OpenSSL sounds right (as well as nudging upstream about supporting OS X )

**Comment by apjanke** [comment]
Closing this due to inactivity/lack of response. 

If you get this working with OpenSSL, feel free to re-submit on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core).

**Pull request closed without merging**